gateway-messages/src/sp_to_mgs.rs

```diff
@@ -199,6 +233,21 @@ pub struct SpStateV2 {
     pub rot: Result<RotStateV2, RotError>,
 }
 
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub struct SpStateV3 {
+    pub hubris_archive_id: [u8; 8],
+    // Serial and revision are only 11 bytes in practice; we have plenty of room
+    // so we'll leave the fields wider in case we grow it in the future. The
+    // values are 0-padded.
+    pub serial_number: [u8; 32],
+    pub model: [u8; 32],
+    pub revision: u32,
+    pub base_mac_address: [u8; 6],
+    pub power_state: PowerState,
+}
```

Comment:
What's the reasoning behind decoupling RoT state from the SP? To allow changes in the future?

Reply:
Going down this path and allowing for cases where either the SP or the RoT had the new version, I could see an exponential growth. Not so bad when the exponent is only 2, but why not decouple?

Reply:
Although our model is that you update RoT first, then SP, we can't really promise  that that will always be the case.